TTHAnalysis/python/plotter/makeShapeCards.py

```diff
@@ -124,19 +126,19 @@ def rebin2Dto1D(h,funcstring):
     newh.SetLineColor(h.GetLineColor())
     return newh
 
-myout = outdir+"/common/" if len(masses) > 1 else outdir;
+myout = os.path.join(options.outdir,"common") if len(masses) > 1 else options.outdir;
 
 report={}
 if options.infile!=None:
-    infile = ROOT.TFile(myout+binname+".bare.root","read")
+    infile = ROOT.TFile(os.path.join(myout,binname+".bare.root"),"read")
     for p in mca.listSignals(True)+mca.listBackgrounds(True)+['data']:
         h = infile.Get(p)
         if h: report[p] = h
 else:
     report = mca.getPlotsRaw("x", args[2], args[3], cuts.allCuts(), nodata=options.asimov)
 
 if options.savefile!=None:
-    savefile = ROOT.TFile(myout+binname+".bare.root","recreate")
+    savefile = ROOT.TFile(os.path.join(myout,binname+".bare.root"),"recreate")
     for n,h in report.iteritems(): savefile.WriteTObject(h,n)
```

Comment:
The code failed because it didn't create the output directory. After fixing that, it didn't work properly with `myout+binname+".root"` when the `myout` didn't end with a slash. So I changed that to use `os.path.join` which will work in all cases. That might not be a critical part of the functionality of this script, but it was annoying to run without.

As you point out, this code is used not by you and me alone, which makes it even more important to fix small issues and to make sure the code is understandable to someone who didn't write it themselves (which is not the case, as it is now). If you value the purity of the git history more than the functionality and usability of the code, I'm happy to agree to disagree with you.

Finally, I'd much rather focus on writing code than on writing long comments arguing about irrelevant details in said code. Especially if the argument is conducted in a rude and condescending manner.